llvm/test/CodeGen/NVPTX/kernel-param-align.ll

```diff
@@ -0,0 +1,36 @@
+; RUN: llc < %s -march=nvptx64 -mcpu=sm_72 2>&1 | FileCheck %s
+; RUN: %if ptxas %{ llc < %s -march=nvptx64 -mcpu=sm_72 | %ptxas-verify %}
+
+%struct.Large = type { [16 x double] }
+
+; CHECK-LABEL: func_align
+; CHECK: .param .u64 .ptr .global .align 16 func_align_param_0,
+; CHECK: .param .u64 .ptr .global func_align_param_1,
+; CHECK: .param .u32 .ptr .global func_align_param_2
```

Comment:
This is still wrong. `addrspace(3)` is *not* a global pointer. It's a shared AS pointer.

Speaking of which, please also add a test for explicitly annotated const and global AS pointers.